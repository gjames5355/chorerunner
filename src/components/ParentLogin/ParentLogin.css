/* 
font-family: 'Press Start 2P', cursive;
font-family: 'Fredoka One', cursive;
font-family: 'Concert One', cursive;
font-family: 'Electrolize', sans-serif; */

.parent-login {
  height: 85vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 15vh;
  overflow-y: auto;
}

.parent-login>h2 {
  color: rgb(255, 217, 0);
  font-size: 30px;
  text-shadow: -1px -1px rgb(105, 90, 0), 1.5px 1.5px rgb(134, 112, 0);
}

.parent-form-container {
  width: 70%;
  padding-bottom: 20px;
}

.parent-form-container div {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 15px 0;
}

.parent-login label {
  font-family: 'Press Start 2P', cursive;
  color: rgb(255, 217, 0);
  padding-bottom: 10px;
}

div[role="alert"]>p {
  color: white;
  width: auto;
  margin: auto;
  font-size: 1.1rem;
  font-weight: 500;
}

input {
  font-size: 17px;
  padding-left: 25px;
  padding-right: 15px;
  color: rgb(105, 90, 0);
  font-family: 'Electrolize', sans-serif;
  margin-bottom: 20px;
  height: 30px;
  border-radius: 15px;
  background-color: white;
  border: 0;
}

.parent-login button {
  width: 120px;
  text-align: center;

  margin: 10px;
  font-family: 'Press Start 2P', cursive;
  font-size: 20px;
  padding: 7px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725F00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 .3em rgb(134, 112, 0);
  cursor: pointer;
}

.demo {
  width: 260px;
  margin: 0 auto;
  display: flex;
  font-family: 'Electrolize', sans-serif;
  color: white;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0px 10px 30px 10px;
  background: rgb(0, 0, 0, .5);
  position: relative;
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
  border-radius: 10px;
}

.demo p {
  margin: 0;
}

@media (min-width: 575px) {
  .parent-form-container {
    width: 40%;
  }
}

@media (min-width: 875px) {
  .parent-form-container {
    width: 320px;
  }
}