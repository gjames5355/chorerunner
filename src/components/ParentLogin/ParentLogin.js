import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import UserContext from '../../contexts/UserContext';
import './ParentLogin.css';

export default class ParentLogin extends Component {
  static contextType = UserContext;

  state = {
    username: '',
    password: '',
    error: null,
  };

  onChangeHandle = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  onLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/parent-dashboard';
    history.push(destination);
  };

  handleSubmit = e => {
    e.preventDefault();
    const { username, password } = e.target;

    this.setState({ error: null });

    AuthApiService.postLogin({
      username: username.value,
      password: password.value,
    })
      .then(res => {
        username.value = '';
        password.value = '';
        this.context.processLogin(res.authToken, res.type);
        this.onLoginSuccess();
        this.setState({ error: null });
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };
  render() {
    const { username, password, error } = this.state;
    return (
      <div className='parent-login container'>
        <h2>Parent Login</h2>

        <div className='demo'>
          <h3>Demo:</h3>
          <p>username: margeincharge</p>
          <p>password: Password123!</p>
        </div>
        <form className='parent-form-container' onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor='username'>Username</label>
            <input
              name='username'
              type='text'
              onChange={this.onChangeHandle}
              value={username}
              required
            ></input>
          </div>
          <div>
            <label htmlFor='password'>Password</label>
            <input
              name='password'
              type='password'
              onChange={this.onChangeHandle}
              value={password}
              required
            ></input>
          </div>
          <div role='alert'>{error && <p>{error}</p>}</div>
          <button type='submit'>login</button>
        </form>
      </div>
    );
  }
}
