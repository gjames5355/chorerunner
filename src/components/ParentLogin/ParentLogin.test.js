import React from 'react';
import ReactDOM from 'react-dom';
import ParentLogin from './ParentLogin';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('ParentLogin renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <ParentLogin />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Parent Login as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ParentLogin />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
