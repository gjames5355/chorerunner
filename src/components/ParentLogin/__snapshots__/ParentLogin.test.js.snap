// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Parent Login as expected 1`] = `
<div
  className="parent-login container"
>
  <h2>
    Parent Login
  </h2>
  <div
    className="demo"
  >
    <h3>
      Demo:
    </h3>
    <p>
      username: margeincharge
    </p>
    <p>
      password: Password123!
    </p>
  </div>
  <form
    className="parent-form-container"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      role="alert"
    />
    <button
      type="submit"
    >
      login
    </button>
  </form>
</div>
`;
